Tooling update

rodelta opened the pull request:
- Organizes the package.json fields (arbitrary convention, can revert)
- Updates storybook to v7 
- Updates typescript to v5
- Sets up vite builder for storybook 
- Updates some existing stories
- Adds a type-checking command, cleans up storybook issues

mnlfischer commented:
Please add a commit with the semantic release convention, otherwise semantic release doesn't know how to bump the version. You can also add an empty commit with the convention. E.g.` feat(<context>): <message>`

rodelta commented:
> Please add a commit with the semantic release convention, otherwise semantic release doesn't know how to bump the version. You can also add an empty commit with the convention. E.g.` feat(<context>): <message>`

I've rebased on the latest changes, changed commit names, and expanded the example stories a bit, (and fixed some type errors)

rodelta commented on src/components/menu/menu.stories.tsx:
```diff
@@ -14,7 +14,7 @@ export const Default: Story = {
     <div className="w-52">
       <Menu>
-        <Menu.Button>Open Menu</Menu.Button>
+        <Menu.Button className="IShouldBeOptional">Open Menu</Menu.Button>
```
This is a fix for a type issue. 
The component `Menu.Button` has the `className` prop as non-optional, so the story requires the thing to be there. I added it to the story to avoid modifying "prod" code, but if it's safe, I can also modify the code and remove it from here.

rodelta commented on src/components/icon-button/icon-button.stories.tsx:
```diff
@@ -27,10 +27,10 @@ export const Default: Story = {
+    //   options: ["chat"],
+    //   control: "text",
+    //   defaultValue: "chat",
+    // },
```
It's giving a type error when I leave it in. 
I didn't modify prod code, but did change the stories to avoid issues.
I'll remove this part of the code for the story, and we can re-add what makes sense if we re-visit this stories file.

rodelta replied:
I've modified the story to use SVG icons, to align it with the interface. It's properly being displayed in Storybook now :)

mnlfischer replied:
great, thanks!
